pkg/metrics/composite_tags.go

```diff
@@ -0,0 +1,115 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package metrics
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+// CompositeTags stores two tag sets and provides methods to manipulate them easily
```

Comment:
It's not clear from this description why I'd want to use this type.

A few other things to mention here:
 * tags aren't unique or in a guaranteed order
 * whether the slices should be considered immutable

Reply:
> It's not clear from this description why I'd want to use this type.

This is a specialized type for carrying tags from aggregator to serializer, and probably shouldn't be used anywhere else.

Reply:
I have updated the description.